#Pandas Basics

**Roadmap:** Introduction to Pandas → Loading & Inspecting Data → Selecting Columns → `loc` vs `iloc` → ✍️ Hands-on 03 → 🎯 Quiz 1

📂 **Dataset:** `tips.csv` — put it in the same folder as this notebook.

## 1️⃣ Introduction to Pandas

> **Flow:** Pandas = NumPy + labels → Series (1D + index) → DataFrame (2D table) → creating from a dict → NumPy sits underneath

In [ ]:
! pip3 install pandas

In [1]:
# import pandas as pd
import pandas as pd
import numpy as np 

In [2]:
# pd.Series() — 1D data with an index
total_bill=[4700,8000,1200,500]
s=pd.Series(total_bill,index=['mon','tue','wed','fri'])
print(s)
print(type(s))

mon    4700
tue    8000
wed    1200
fri     500
dtype: int64
<class 'pandas.core.series.Series'>


In [ ]:
# pd.DataFrame() — from a dictionary

restaurant_data = {
    'Table_No': [12, 14, 15],
    'Bill_Amount': [1250, 890, 2400],
    'Waiter_Name': ['Amit', 'Priya', 'Rahul']
}
df = pd.DataFrame(restaurant_data)

# print(df)
# print(type(df))
print(df['Bill_Amount'])
print(type(df['Bill_Amount']))

In [ ]:
# .values / .to_numpy() — the NumPy array underneath

print(df.values)
print(type(df.values))

### 🧩 Try these — Section 1 

**1.1** Make a `Series` containing 4 late-night food delivery items (as labels) and their total orders (as data): 'Pizza' 45, 'Burger' 30, 'Pasta' 15, 'Fries' 60. Then, print the total orders for 'Pizza' using its label, not its numerical position.

**1.2** Create a Python dictionary and use it to build a `DataFrame`. It should have these three columns:
* `Brand` (e.g., Nike, Puma, Campus)
* `Price` (their respective amounts)
* `In_Stock` (True or False)


## 2️⃣ Loading & Inspecting Data

> **Flow:** `pd.read_csv` → `head()` / `tail()` → `shape`, `columns` → `info()` (dtypes + nulls together) → `describe()` → `value_counts()` → `unique()` / `nunique()`

In [4]:

df=pd.read_csv("/Users/shailesh/Desktop/vedam_Sem3_ml/Data/tip.csv")


In [5]:
# head(), tail()
# print(df.head(20))
print(df.tail(10))


     total_bill   tip     sex smoker   day    time  size
234       15.53  3.00    Male    Yes   Sat  Dinner     2
235       10.07  1.25    Male     No   Sat  Dinner     2
236       12.60  1.00    Male    Yes   Sat  Dinner     2
237       32.83  1.17    Male    Yes   Sat  Dinner     2
238       35.83  4.67  Female     No   Sat  Dinner     3
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2


In [8]:
df.sample(5)

,total_bill,tip,sex,smoker,day,time,size
55,19.49,3.51,Male,No,Sun,Dinner,2
181,23.33,5.65,Male,Yes,Sun,Dinner,2
201,12.74,2.01,Female,Yes,Thur,Lunch,2
139,13.16,2.75,Female,No,Thur,Lunch,2
38,18.69,2.31,Male,No,Sat,Dinner,3


In [ ]:
# shape, columns

df.shape
print(df.columns)


In [9]:
# info() — dtypes and null counts in one shot
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB


In [10]:
# describe() — statistics of numeric columns
df.describe()

,total_bill,tip,size
count,244.000000,244.000000,244.000000
mean,19.785943,2.998279,2.569672
std,8.902412,1.383638,0.951100
min,3.070000,1.000000,1.000000
25%,13.347500,2.000000,2.000000
50%,17.795000,2.900000,2.000000
75%,24.127500,3.562500,3.000000
max,50.810000,10.000000,6.000000


In [11]:
# value_counts(), unique(), nunique()
print(df['day'].value_counts())
print(df['day'].unique())
print(df['day'].nunique())

day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64
['Sun' 'Sat' 'Thur' 'Fri']
4


### 🧩 Try these — Section 2

⏱️ *5–7 minutes.* Everything so far was on `tips`. Now run the same checks on a different dataset — the columns change, the method doesn't.

**2.1** Load the Titanic dataset into a DataFrame called `t`. Print its shape — how many rows and how many columns?

**2.2** Run `t.info()`. Some columns have fewer values than the total number of rows. Write down which columns those are. ⭐

**2.3** Run `t.describe()`. It shows fewer columns than `info()` listed. Write down one column it left out, and why.

**2.4** In `t`, find how many different values the `class` column has, list what those values are, and print how many passengers fall into each one. Three lines. ⭐